### DISCUSSION TITLE
Next.js API and DB Connections for Typeorm and Mongo

### DISCUSSION THREAD

**Discussion by hamdongunner** [vercel/next.js#14550] [Help]
Should I reconnect to the DB in every handler when I use next.js for API? Or is there a better way to do that?

And isn't that a bad thing if it was the regular approach?

I tried to create a nextjs project with typeorm or MongoDB and all the examples and resources look like they are waiting for the connection with each call.

This [image](https://i.stack.imgur.com/bvZAm.png) is for typeorm example. I got it from this [repo](https://github.com/mthomps4/next-now-test/tree/next-typeorm-example)

And here is the Mongo articles that I found that they instruct to add the DB connection in middleware and recall it in every handler.

https://developer.mongodb.com/how-to/nextjs-building-modern-applications https://hoangvvo.com/blog/full-fledged-app-with-next-js-and-mongodb-part-1/

and here is the repos middleware to check it too if you like

https://github.com/kukicado/building-modern-app-with-nextjs-and-mongodb/blob/master/mct/middleware/database.js

https://github.com/hoangvvo/nextjs-mongodb-app/blob/master/middlewares/database.js

Couldn't I just create an entry point to set up my connection for one time so I can keep using it without re-waiting everytime?

**Comment by hanselabreu** [comment]
I agree with what you're saying, it would be ideal to establish the connection only once. Ideally at startup and not when a route is hit.
